arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -1325,6 +1328,160 @@
+            "$ref": "#/definitions/RouteProperties"
+          }
+        },
+        "fallbackRoute": {
+          "description": "The properties of the route that will be used as a fallback route when none of the conditions specified in the 'routes' section are met.",
+          "$ref": "#/definitions/RouteProperties"
+        }
+      }
+    },
+    "RoutingEndpoints": {
+      "description": "The properties related to the user-provided endpoints to which IoT hub routes the messages to, based on the routing rules. A maximum of 10 endpoints is allowed across all endpoint types for paid hubs and a maximum of 1 endpoint is allowed across all endpoint types for free hubs.",      
+      "type": "object",
+      "properties": {
+        "serviceBusQueues": {
+          "description": "The list of service bus queue endpoints to which IoT hub routes the messages to, based on the routing rules.",      
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RoutingMessagingEndpointProperties"
+          }
+        },
+        "serviceBusTopics": {
+          "description": "The list of service bus topic endpoints to which IoT hub routes the messages to, based on the routing rules.",      
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RoutingMessagingEndpointProperties"
+          }
+        },
+        "eventHubs": {
+          "description": "The list of eventhub endpoints to which IoT hub routes the messages to, based on the routing rules. This list should not include the in-built eventhub endpoint.",      
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RoutingEventHubProperties"
+          }
+        }
+      }
+    },
+    "RoutingMessagingEndpointProperties": {
+      "description": "The properties related to service bus queue or topic endpoint types.",
+      "type": "object",
+      "properties": {
+        "connectionString": {
+          "description": "The connectionstring of the service bus queue or topic endpoint.",
+          "type": "string"
+        },
+        "name": {
+          "description": "The name of the service bus queue or topic endpoint. The endpoint name cannot be null or whitespace and should not start with a '$'. The endpoint name should not exceed 64 characters. The following names are reserved;  events, operationsMonitoringEvents, fileNotifications, $default. Endpoint names have to be unique across endpoint types.",
+          "type": "string",
+          "pattern": "^[A-Za-z0-9-._]{1,64}$"
+        },
+        "id": {
+          "description": "The identifier of the endpoint. This will be set by the server. Modifying this value causes create or update IoT hub operations to fail. Pass id as null when specifying a new endpoint. Pass the same id that you get for an endpoint from the service when specifying an existing endpoint during update operations.",
+          "type": "string",
+          "readOnly": true
```

Comment:
Will remove the readOnly property. I was not aware that it would actually remove the property being passed in. Was considering it as an indication to the user to not modify the value. Also, we are aware and debated at length regarding the weirdness with this behavior. However, there are reasons for doing it the way it is being done. We can have a 1:1 conversation if you care to understand. Btw, who is the "API review board" ?